SpottedBy crashes because of nil pointer dereferences

**Describe the bug**
When requesting the list of players that a given player has spotted then the program crashes because of a nil pointer dereference.

**To Reproduce**
Request all players that a given player has spotted. I've added a (unnecessary) nil pointer check to demonstrate that the event spotter is not nil.
It happens for any demo that I tried (tried 5 different official premier ones).

Code:
```go
func main() {
	if err := demoinfocs.ParseFile("file_path", func(p demoinfocs.Parser) error {
		p.RegisterEventHandler(func(e events.PlayerSpottersChanged) {
			if e.Spotted != nil {
				p.GameState().Participants().SpottedBy(e.Spotted)
			}

		})
		return nil
	}); err != nil {
		panic(err)
	}
}

```

**Expected behavior**
Get a list of the spotted by players

**Library version**
v5.1.0

**Additional context**
The relevant part of the crash logs
```
panic: failed to parse demo: runtime error: invalid memory address or nil pointer dereference
        stacktrace:
        goroutine 21 [running]:
        runtime/debug.Stack()
                /home/vincent/.asdf/installs/golang/1.25.5/go/src/runtime/debug/stack.go:26 +0x5e
        github.com/markus-wa/demoinfocs-golang/v5/pkg/demoinfocs.NewParserWithConfig.func1({0x7a3a20, 0xc001614080})
                /home/vincent/.asdf/installs/golang/1.25.5/packages/pkg/mod/github.com/markus-wa/demoinfocs-golang/v5@v5.1.0/pkg/demoinfocs/pa
rser.go:552 +0x2a
        github.com/markus-wa/godispatch.(*Dispatcher).dispatchWithRecover.func1()
                /home/vincent/.asdf/installs/golang/1.25.5/packages/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:197 +0x3b
        panic({0x7a3a20?, 0xc001614080?})
                /home/vincent/.asdf/installs/golang/1.25.5/go/src/runtime/panic.go:783 +0x132
        github.com/markus-wa/godispatch.callConsumerCode.func1()
                /home/vincent/.asdf/installs/golang/1.25.5/packages/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:122 +0x4d
        panic({0x7a3a20?, 0xc0015f2000?})
                /home/vincent/.asdf/installs/golang/1.25.5/go/src/runtime/panic.go:783 +0x132
        github.com/markus-wa/godispatch.callConsumerCode.func1()
                /home/vincent/.asdf/installs/golang/1.25.5/packages/pkg/mod/github.com/markus-wa/godispatch@v1.4.1/dispatch.go:122 +0x4d
        panic({0x760d80?, 0xbc56c0?})
                /home/vincent/.asdf/installs/golang/1.25.5/go/src/runtime/panic.go:783 +0x132
        github.com/markus-wa/demoinfocs-golang/v5/pkg/demoinfocs/common.(*Player).IsSpottedBy(0xc000318f30?, 0x0?)
                /home/vincent/.asdf/installs/golang/1.25.5/packages/pkg/mod/github.com/markus-wa/demoinfocs-golang/v5@v5.1.0/pkg/demoinfocs/co
mmon/player.go:212 +0x35
        github.com/markus-wa/demoinfocs-golang/v5/pkg/demoinfocs.participants.SpottedBy(...)
                /home/vincent/.asdf/installs/golang/1.25.5/packages/pkg/mod/github.com/markus-wa/demoinfocs-golang/v5@v5.1.0/pkg/demoinfocs/ga
me_state.go:463
```

--- a/pkg/demoinfocs/common/player.go
+++ b/pkg/demoinfocs/common/player.go
@@ -202,6 +202,10 @@ func (p *Player) IsSpottedBy(other *Player) bool {
 	if p.Entity == nil {
 		return false
 	}
+	pawnEntity := p.PlayerPawnEntity()
+	if pawnEntity == nil {
+		return false
+	}
 
 	clientSlot := other.EntityID - 1
 	bit := uint(clientSlot)
@@ -209,10 +213,10 @@ func (p *Player) IsSpottedBy(other *Player) bool {
 	var mask st.Property
 
 	if bit < 32 {
-		mask = p.PlayerPawnEntity().Property("m_bSpottedByMask.0000")
+		mask = pawnEntity.Property("m_bSpottedByMask.0000")
 	} else {
 		bit -= 32
-		mask = p.PlayerPawnEntity().Property("m_bSpottedByMask.0001")
+		mask = pawnEntity.Property("m_bSpottedByMask.0001")
 	}
 
 	return (mask.Value().UInt64() & (1 << bit)) != 0

